File: Striver Important Interview Questions/Day 16 - String Part 2/valid-anagram.cpp

// https://leetcode.com/problems/valid-anagram/description/

class Solution {
public:
    bool isAnagram(string s, string t) {
        
        int sLen = s.size();
        int tLen = t.size();

        sort(s.begin(), s.end());
        sort(t.begin(), t.end());

        return (sLen == tLen) && (s == t);
    }
};

File: Neetcode-150-js/Arrays and Hashing/valid-anagram.js

// https://leetcode.com/problems/valid-anagram/description/

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

function sortString(s){
    return s.split("").sort().join("");
}

var isAnagram = function(s, t) {
    let sSorted = sortString(s);
    let tSorted = sortString(t);
    return sSorted === tSorted;
};